### DISCUSSION TITLE
Get access to notifications since the removal of useNotifications hook

### DISCUSSION THREAD

**Discussion by jt561** [mantinedev/mantine#5192] [Q&A] [answered]
Hello, I'm a little bit late but I just upgraded to version 6, and have an implementation that would take the notifications from mantines notification system. The implementation used `useNotifications` and would stre any new notification in mongoDB. 

My question is, what is the alternative to getting all notification that are created with `notifications.show` or `notifications.update`?

**Comment by rtivital** [comment]
No, version 6 does not support that. You need to track it on your side.

**Reply by jt561** [reply]
Ok thank you

**Comment by jt561** [comment]
If anyone has any suggestions on how I can track it from my side, please do let me know

**Reply by jt561** [reply]
The solution I found: 
```tsx
import { useEffect, useState } from "react";
import { useNotifications, NotificationData, notificationsStore } from "@mantine/notifications";

export default function NotificationsX() {
    const { notifications: mantineNotifications } = useNotifications(notificationsStore);
    const [mantineNotificationsHistory, setMantineNotificationsHistory] = useState<NotificationData[]>([]);

    useEffect(() => {
        setMantineNotificationsHistory((prev) => {
            const newNotifications: NotificationData[] = mantineNotifications.filter(
                (mantineNotification) => !prev.find((historyNotification) => historyNotification.id === mantineNotification.id)
            );
            return [...newNotifications, ...prev];
        });

        // eslint-disable-next-line
    }, [mantineNotifications]);

    return (
        <>
               {mantineNotificationsHistory.map((historyNotification) => (
                    <div>{JSON.stringify(historyNotification)}</div>
               )}
        </>
    );
}
```

Pretty simple way to get all notifications created by Mantine.

I think it'll worth mentioning `notificationsStore` and `useNotifications` in the notifications docs. I believe it'll be very helpful to others.

**Comment by jt561** [comment, marked as answer]
[Pretty simple solution](https://github.com/orgs/mantinedev/discussions/5192#discussioncomment-7655998)
